services/migrations/dump.go

```
@@ -395,12 +398,13 @@ func (g *RepositoryDumper) createItems(dir string, itemFiles map[int64]*os.File,
 			itemFiles[number] = itemFile
 		}
 
-		bs, err := yaml.Marshal(items)
-		if err != nil {
+		encoder := yaml.NewEncoder(itemFile)
+		if err := encoder.Encode(items); err != nil {
+			_ = encoder.Close()
 			return err
 		}
 
-		if _, err := itemFile.Write(bs); err != nil {
+		if err := encoder.Close(); err != nil {
```

1. Why can a `Close` method return an error?
2. Why do we pass this error down? I mean, the function already completed what it is supposed to do, why let it fail afterward?
3. Couldn't we do `encoder := yaml.NewEncoder(itemFile); defer encoder.Close(); return encoder.Encode(items)` as well?